### DISCUSSION TITLE
Jinja Templating IDE, Extensions and Tips

### DISCUSSION THREAD

**Discussion by tlourey** [nautobot/nautobot#5074] [Q&A] [answered]
I'm wondering what IDE/Editor and extensions/plugins (for said IDE) do others use for their jinja template creation? 

Also any tricks to help speed up their creation?

https://[textfsm.nornir.tech](https://textfsm.nornir.tech/)/ when set to Jinja but in Visual Studio ideally but i'm open to notepad++ suggestions as well. vim as a last resort...

**Comment by jdrew82** [comment]
Most of us are VS Code users and just use the extensions for that App. I'd advise using one of the many Jinja2 templating Apps out there like Template Designer For All or JinjaFX. Both have self contained deployments that make it easy to work with templates and variables to test template generation.

**Reply by tlourey** [reply]
I'm a VS Code user, but I was hoping there was something like JinjaFX for VS Code. The extensions I've seen so far don't seem to help much other than syntax highlighting. What extensions do you use? 

Do you have a link for Template Designer For All, I'm struggling to find it. Thanks

**Reply by jdrew82** [reply, marked as answer]
My mistake, it's Template Designer for Automation! It's [here](https://github.com/cidrblock/td4a). And here's [JinjaFX](https://github.com/cmason3/jinjafx). Regarding extensions that add that capability into VS Code, unfortunately, I'm not aware of any.

**Reply by tlourey** [reply]
Many thanks

**Comment by tlourey** [comment]
For anyone interested, I found a VSCode extension called Template Finder that kind of works if you use YAML files instead of JSON as the variables. It only works in simple cases as it doesn't support Jinja Notation. Still for beginners or inital simple development its a *little* bit helpful. There is [feature request](https://github.com/ArtyMaury/template-finder/issues/23) to improve its support. 

[VS Code Marketplace Link](https://marketplace.visualstudio.com/items?itemName=Artymaury.template-finder)
